lib/http2/connection.c

```diff
@@ -614,6 +618,14 @@ static int handle_priority_frame(h2o_http2_conn_t *conn, h2o_http2_frame_t *fram
         if (h2o_http2_scheduler_get_weight(&stream->_refs.scheduler) != 257)
             set_priority(conn, stream, &payload, 1);
     } else {
+        /* Ignore PRIORITY frames for closed streams */
+        if (frame->stream_id & 1) {
```

Comment:
Could you please change the check to `if (h2o_http2_stream_is_push(frame->stream_id))`? That is the wrapper function we have.